dex/btc/clone.go

```diff
@@ -0,0 +1,39 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package btc
+
+import (
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/wire"
+)
+
+// ReadCloneParams translates a CloneParams into a btcsuite chaincfg.Params.
+func ReadCloneParams(cloneParams *CloneParams) *chaincfg.Params {
+	return &chaincfg.Params{
+		PubKeyHashAddrID: cloneParams.PubKeyHashAddrID,
+		ScriptHashAddrID: cloneParams.ScriptHashAddrID,
+		Bech32HRPSegwit:  cloneParams.Bech32HRPSegwit,
+		CoinbaseMaturity: cloneParams.CoinbaseMaturity,
+		Net:              wire.BitcoinNet(cloneParams.Net),
+	}
```

Comment:
@chappjc: What do you think of this instead? Clones can just use `CloneParams`.

Reply:
Seems pretty straightforward to use `dex/btc.ReadCloneParams` that way.  Of course this means that `dex/btc.ReadCloneParams` needs to be updated for any clone that wants to use a field of `chaincfg.Params` that isn't already in `CloneParams`.  It also means that clones that don't change one of these params still have to specify it, but that's not much of a burden IMO.  So yeah, this is fine with me.

Reply:
Unless I'm reading this wrong, it's a little backwards. Clones **must** use all of the fields, and populating extra fields would never change anything. The underlying calls are always to **btcd**'s packages.

Reply:
Say if BTCX wanted to change `PrivateKeyID`, that field would need to be added to the `CloneParams` struct and `ReadCloneParams` modified to copy it.  That seemed like the benefit to marshal/unmarshal is that you could set anything and have it update.
But if you have to add a field to CloneParams for BTCX, then server/asset/ltc needs to update their CloneParams definitions as well lest they get the zero value for that new field copied over instead of the default/BTC one.